internal/api/v1/dto/favourites.go

```
@@ -0,0 +1,100 @@
+	AssetTypeInsight  = favourites.AssetTypeInsight
+	AssetTypeAudience = favourites.AssetTypeAudience
+)
+
+// ChartPayload represents the chart-specific payload exposed over HTTP.
+type ChartPayload struct {
+	Title string    `json:"title"`
+	XAxis string    `json:"xAxis"`
+	YAxis string    `json:"yAxis"`
+	Data  []float64 `json:"data"`
+}
+
+// InsightPayload represents the insight-specific payload.
+type InsightPayload struct {
+	Text string `json:"text"`
+}
+
+// AudiencePayload represents the audience-specific payload.
+type AudiencePayload struct {
+	Gender                   string `json:"gender"`
+	BirthCountry             string `json:"birthCountry"`
+	AgeGroup                 string `json:"ageGroup"`
+	HoursOnSocialMediaPerDay int    `json:"hoursOnSocialMediaPerDay"`
+	PurchasesLastMonth       int    `json:"purchasesLastMonth"`
+}
+
+// CreateAssetRequest represents an asset to be created inline.
+type CreateAssetRequest struct {
+	Type        AssetType        `json:"type"`
+	Description string           `json:"description"`
+	Chart       *ChartPayload    `json:"chart,omitempty"`
+	Insight     *InsightPayload  `json:"insight,omitempty"`
+	Audience    *AudiencePayload `json:"audience,omitempty"`
+}
+
+// CreateFavouriteRequest is the request body for creating a new favourite asset.
+type CreateFavouriteRequest struct {
+	Asset CreateAssetRequest `json:"asset"`
+}
+
+// EditFavouriteDescriptionRequest is the body for updating a favourite's description.
+type EditFavouriteDescriptionRequest struct {
+	Description string `json:"description"`
+}
+
+// AssetResponse is the asset representation returned by the API.
+type AssetResponse struct {
+	ID          string           `json:"id"`
+	Type        AssetType        `json:"type"`
+	Description string           `json:"description"`
+	Chart       *ChartPayload    `json:"chart,omitempty"`
+	Insight     *InsightPayload  `json:"insight,omitempty"`
+	Audience    *AudiencePayload `json:"audience,omitempty"`
```

Could there be a case where more than one is set?

---

Good catch. Right now the mapper uses the field that matches the type on the model and silently ignores any extras.
Even that the domain model enforces a single variant, it would be better to validate this at the DTO/mapper level and early reject payloads when more than one is set.